### PR TITLE
[FLINK-20953][canal][json] Option 'canal-json.database.include' and 'canal-json.table.include' support regular expression

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*Currently Canal format option 'canal-json.database.include' and 'canal-json.table.include' don't support regular expression to match. It's necessary to use regular expression to match the field 'database' and 'table' of Canal binlog record.*

## Brief change log

  - *`CanalJsonDeserializationSchema` matches the field 'database' and 'table' of Canal binlog message using regular expression for Canal format option 'canal-json.database.include' and 'canal-json.table.include'.*

## Verifying this change

  - *`CanalJsonSerDeSchemaTest` modifies the `testFilteringTables` test case to use regular expression for option 'canal-json.database.include' and 'canal-json.table.include' to verify the regular matching of the field 'database' and 'table' of Canal binlog record.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/canal/CanalJsonDeserializationSchema.java [hunk 1]
```diff
@@ -197,12 +198,12 @@ public void deserialize(@Nullable byte[] message, Collector<RowData> out) throws
         try {
             final JsonNode root = jsonDeserializer.deserializeToJsonNode(message);
             if (database != null) {
-                if (!database.equals(root.get(ReadableMetadata.DATABASE.key).asText())) {
+                if (!Pattern.matches(database, root.get(ReadableMetadata.DATABASE.key).asText())) {
```
Comment (wuchong) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/canal/CanalJsonDeserializationSchema.java:201]: We should reuse the pattern.

**Comment by SteNicholas** [comment]
@wuchong, I have reused the pattern for the specific database and table. Please review again for this minor change.

File: docs/dev/table/connectors/formats/canal.md [hunk 2]
```diff
@@ -292,14 +292,14 @@ Format Options
       <td>optional</td>
       <td style="word-wrap: break-word;">(none)</td>
       <td>String</td>
-      <td>Only read changelog rows which match the specific database (by comparing the "database" meta field in the Canal record).</td>
+      <td>Only read changelog rows which match the specific database (by regular matching the "database" meta field in the Canal record).</td>
```
Comment (wuchong) [inline docs/dev/table/connectors/formats/canal.md:295]: What do you think about improving the description to the following?

```
An optional regular expression to only read the speicifc databases changelog rows by regular matching the "database" meta field in the Canal record. The pattern string is compatible with Java's <a href="https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html">Pattern</a>.
```

Would be great if you can update the description in `org.apache.flink.formats.json.canal.CanalJsonOptions#DATABASE_INCLUDE`.

**Review by wuchong — COMMENTED** [review]
Thanks. The change looks good to me. 

I only left a minor comment for the docs.

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**